helm/nessie/templates/_helpers.tpl

```diff
@@ -158,7 +158,9 @@ Apply S3 catalog options.
 {{- if .pathStyleAccess -}}{{- $_ := set $map ( print $prefix "path-style-access" ) .pathStyleAccess -}}{{- end -}}
 {{- if .accessPoint -}}{{- $_ := set $map ( print $prefix "access-point" ) .accessPoint -}}{{- end -}}
 {{- if .allowCrossRegionAccessPoint -}}{{- $_ := set $map ( print $prefix "allow-cross-region-access-point" ) .allowCrossRegionAccessPoint -}}{{- end -}}
-{{- if .clientAuthenticationMode -}}{{- $_ := set $map ( print $prefix "auth-mode" ) .clientAuthenticationMode -}}{{- end -}}
+{{- if .serverAuthenticationMode -}}{{- $_ := set $map ( print $prefix "server-auth-mode" ) .serverAuthenticationMode -}}{{- end -}}
+{{- if .clientAuthenticationMode -}}{{- $_ := set $map ( print $prefix "client-auth-mode" ) .clientAuthenticationMode -}}{{- end -}}
```

Comment:
Not porting over the "legacy" `auth-mode` property since this is still more or less in beta. If someone disagrees I can add it.